File: Intermediate/14.Implement a stack and its operations (push, pop, peek) using an array/Solution.cpp
#include <iostream>
#define MAX 100
using namespace std;

class Stack {
    int top;
    int arr[MAX];

public:
    Stack() : top(-1) {}

    void push(int value) {
        if (top == MAX - 1) {
            cout << "Stack overflow!" << endl;
        } else {
            arr[++top] = value;
            cout << value << " pushed into the stack." << endl;
        }
    }

    void pop() {
        if (top == -1) {
            cout << "Stack underflow!" << endl;
        } else {
            cout << arr[top--] << " popped from the stack." << endl;
        }
    }

    void peek() {
        if (top == -1) {
            cout << "Stack is empty." << endl;
        } else {
            cout << "Top element is: " << arr[top] << endl;
        }
    }
};

int main() {
    Stack stack;
    int choice, value;

    do {
        cout << "\n1. Push\n2. Pop\n3. Peek\n4. Exit\nEnter your choice: ";
        cin >> choice;

        switch (choice) {
        case 1:
            cout << "Enter value to push: ";
            cin >> value;
            stack.push(value);
            break;
        case 2:
            stack.pop();
            break;
        case 3:
            stack.peek();
            break;
        case 4:
            cout << "Exiting..." << endl;
            break;
        default:
            cout << "Invalid choice!" << endl;
        }
    } while (choice != 4);

    return 0;
}

File: Intermediate/14.Implement a stack and its operations (push, pop, peek) using an array/Solution.py
class Stack:
    def __init__(self):
        self.stack = []

    def push(self, value):
        self.stack.append(value)
        print(f"{value} pushed into the stack.")

    def pop(self):
        if not self.stack:
            print("Stack underflow!")
        else:
            print(f"{self.stack.pop()} popped from the stack.")

    def peek(self):
        if not self.stack:
            print("Stack is empty.")
        else:
            print(f"Top element is: {self.stack[-1]}")

if __name__ == "__main__":
    stack = Stack()

    while True:
        print("\n1. Push\n2. Pop\n3. Peek\n4. Exit")
        choice = int(input("Enter your choice: "))

        if choice == 1:
            value = int(input("Enter value to push: "))
            stack.push(value)
        elif choice == 2:
            stack.pop()
        elif choice == 3:
            stack.peek()
        elif choice == 4:
            print("Exiting...")
            break
        else:
            print("Invalid choice!")
